File: services/client/src/protocol/ack_packet.go
package protocol

import (
	"encoding/binary"
	"errors"
)

const ACK_PACKET_LEN int = 1

type AckPacket struct{}

func CreateAckPacket() Packet {
	return &AckPacket{}
}

func (e *AckPacket) GetType() uint8 {
	return TYPE_ACK
}

func (e *AckPacket) Header() []byte {
	return []byte{e.GetType()}
}

func (e *AckPacket) Serialize() []byte {
	message := make([]byte, 0, ACK_PACKET_LEN+LENGTH_BYTES)
	message = binary.BigEndian.AppendUint16(message, uint16(ACK_PACKET_LEN))
	message = append(message, e.Header()...)
	return message
}

func AckFromBytes(bytes []byte) (*AckPacket, error) {
	if len(bytes) != ACK_PACKET_LEN {
		return nil, errors.New("Invalid packet length")
	}
	if bytes[0] != TYPE_ACK {
		return nil, errors.New("Invalid packet type")
	}
	return &AckPacket{}, nil
}

File: services/server/src/protocol/ack_packet.py
from protocol.packet import Packet, TYPE_ACK, ENDIANNESS, LENGTH_BYTES

ACK_PACKET_LEN = 1

class AckPacket(Packet):
    """
    Paquete de tipo ACK
    """
    def __init__(self):
        super().__init__()

    def get_type(self):
        return TYPE_ACK

    def serialize(self):
        message = bytearray()
        length = ACK_PACKET_LEN.to_bytes(LENGTH_BYTES, ENDIANNESS, signed=False)
        message.extend(length)
        message.extend(self.header())
        return message

def ack_from_bytes(bytes: bytes) -> AckPacket:
    if len(bytes) != ACK_PACKET_LEN:
        raise ValueError("Invalid packet length")
    if bytes[0] != TYPE_ACK:
        raise ValueError("Invalid packet type")
    return AckPacket()
